docs: update useWishlist docs (#363)
vuestorefront/bigcommerce: docs/composables/use-wishlist.md

## Features

The `useWishlist` composable is used to:

- create wishlist,
- load wishlist from BigCommerce platform,
- add products to the wishlist,
- remove products from the wishlist.

## API

- `wishlist: Wishlist` - a main data object.

> - `load` - function for getting wishlist from the BigCommerce platform or creating a new one if it doesn't already exist. This method populates the `wishlist` object with the result.

Reviewer: ```suggestion
- `load` - a function for getting a wishlist from the API or creating a new one if it doesn't already exist. This method populates the `wishlist` object with the returned data.
```

> - `addItem` - function for adding products to the wishlist. This method populates the `wishlist` object with the updated result and accepts a single object parameter with the following properties:

Reviewer: ```suggestion
- `addItem` - a function for adding products to the wishlist. This method populates the `wishlist` object with the returned data and accepts a single object parameter with the following properties:
```

  - `product: Product` - See the [Product](../reference/api/bigcommerce-api.product.md) interface for more information.

> - `removeItem` - function for removing a single item from the wishlist. This method populates the `wishlist` object with the updated result and accepts a single object parameter with the following properties:

Reviewer: ```suggestion
- `removeItem` - a function for removing a single item from the wishlist. This method populates the `wishlist` object with the returned data and accepts a single object parameter with the following properties:
```

>   - `product: WishlistItem` - See tthe [WishlistItem](../reference/api/bigcommerce.wishlistitem.md) interface for more information.

Reviewer: ```suggestion
  - `product: WishlistItem` - See the [WishlistItem](../reference/api/bigcommerce.wishlistitem.md) interface for more information.
```

> - `clear` - a function that removes all products from the wishlist. This method populates the `wishlist` object with the updated result.

Reviewer: ```suggestion
- `clear` - a function for removing all products from the wishlist. This method populates the `wishlist` object with the returned data.
```

> - `isInWishlist` - a function that checks if a product is on the wishlist. It returns boolean value. This method accepts a single params object. The params has the following property:

Reviewer: ```suggestion
- `isInWishlist` - a function for checking if a product is on the wishlist. It returns a boolean value. This method accepts a single params object with the following property:
```

  - `product: Product` - See the [Product](../reference/api/bigcommerce-api.product.md) interface for more information.

> - `loading: boolean` - a reactive object containing information about loading state of the wishlist.

Reviewer: ```suggestion
- `loading: boolean` - a reactive object containing information about the loading state of the methods described above.
```

> - `error: UseWishlistErrors` - reactive object containing the error message if any of the methods above failed for any reason. See the [UseWishlistErrors](https://docs.vuestorefront.io/v2/reference/api/core.usewishlisterrors.html) interface for more information.

Reviewer: ```suggestion
- `error: UseWishlistErrors` - a reactive object containing the error message if any of the methods above failed for any reason. See the [UseWishlistErrors](https://docs.vuestorefront.io/v2/reference/api/core.usewishlisterrors.html) interface for more information.
```
